modules/common/namespace/main.tf

```diff
@@ -97,3 +97,30 @@ resource "kubernetes_role_binding" "tiller_role_binding" {
     namespace = kubernetes_namespace.ns[0].metadata[0].name
   }
 }
+
+resource "kubernetes_service_account" "certificate_watcher_service_account" {
+  count = var.enabled ? 1 : 0
+  metadata {
+    name = "certificate-watcher"
+    namespace = kubernetes_namespace.ns[0].metadata[0].name
+  }
+  automount_service_account_token = true
+}
+
+resource "kubernetes_role_binding" "certificate_watcher_role_binding" {
+  count = var.enabled ? 1 : 0
+  metadata {
+    name      = "certificate-watcher-binding"
+    namespace = kubernetes_namespace.ns[0].metadata[0].name
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = "certificate-read"
```

Comment:
I wanted to use variable interpolation here to reference the name of the `ClusterRole`:

https://github.com/liatrio/lead-terraform/blob/75afa96b17b88893fa4a8d58ad8db0d13bae5a36/modules/lead/infrastructure/certmanager.tf#L58

But I wasn't able to figure out a good way to do that.

Reply:
why not move these out of the namespace module and back into the certmanager.tf

Reply:
I believe the service account has to be created within the namespace, otherwise the job in that namespace wouldn't be able to use it.

Since the service account needs to stay here, I think it would be easiest to keep the `RoleBinding` here as well so we can easily interpolate the variables for the service account name.